docs/deployment/getting-started-locally/getting-started-locally.md

```diff
@@ -0,0 +1,196 @@
+
+#### Debug mode
+
+```bash
+> make deploy-debug
+```
+
+This is similar to `make deploy-dev` but additionally configures containers in pods for debugging as required for each container's runtime technology. The associated debugging ports are exposed and labelled so that they can be port-forwarded to the local machine. Skaffold disables automatic image rebuilding and syncing when using the `debug` mode as compared to `dev` mode.
+
+Go debugging uses [Delve](https://github.com/go-delve/delve). Please see the [skaffold debugging documentation](https://skaffold.dev/docs/workflows/debug/) how to setup your IDE accordingly. 
+
+> **Note:** Resuming or stopping only a single goroutine (Go Issue [25578](https://github.com/golang/go/issues/25578), [31132](https://github.com/golang/go/issues/31132)) is currently not supported, so the action will cause all the goroutines to get activated or paused.
+
+This means that when a goroutine is paused on a breakpoint, then all the other goroutines are also paused. This should be kept in mind when using `skaffold debug`.
+
+## 03-Configure Backup [*Optional*]
+
+> **Note:** If you wish to do not backup any snapshots
+
+### Deploying a Local Backup Store Emulator
+
+> **Note:** This section is ***Optional*** and is only meant to describe steps to deploy a local object store which can be used for testing and development. If you either do not wish to enable backups or you wish to use remote (infra-provider-specific) object store then this section can be skipped.
+
+An `Etcd` cluster provisioned via etcd-druid provides a capability to take regular delta and full snapshots which are stored in an object store. You can enable this functionality by ensuring that you fill in [spec.backup.store](https://github.com/gardener/etcd-druid/blob/3383e0219a6c21c6ef1d5610db964cc3524807c8/config/samples/druid_v1alpha1_etcd.yaml#L49-L54) section of the `Etcd` CR. 
+
+| Backup Store Variant          | Setup Guide                                                |
+| ----------------------------- | ---------------------------------------------------------- |
+| Azure Object Storage Emulator | [Manage Azurite](manage-azurite-emulator.md) (Steps 00-03) |
+| S3 Object Store Emulator      | [Manage LocalStack](manage-s3-emulator.md) (Steps 00-03)   |
+
+### Setting up Cloud Provider Object Store Secret
+
+> **Note:** This section is ***Optional***. If you have disabled backup functionality or if you are using local storage or one of the supported object store emulators then you can skip this section.
+
+A Kubernetes [Secret](https://kubernetes.io/docs/concepts/configuration/secret/) needs to be created for cloud provider Object Store access. You can refer to the Secret YAML templates [here](https://github.com/gardener/etcd-backup-restore/tree/master/example/storage-provider-secrets).  Replace the dummy values with the actual configuration and ensure that you have added the `metadata.name` and `metadata.namespace` to the secret.
```

Comment:
We also need to add information about which permissions are to be configured for the acess credentials. For instance, the AWS S3 credential needs to have permissions like `s3:*` (or more restrictive permissions, once we evaluate those correctly for each provider). How would you like to go about this?

Reply:
Sure we can enhance the document once we have finalised the list of permissions that are required. We can then include the changes in the next documentation PR.